pkg/shp/cmd/follower/follow.go

```diff
@@ -117,6 +117,11 @@ func (f *Follower) stop() {
 
 // OnEvent reacts on pod state changes, to start and stop tailing container logs.
 func (f *Follower) OnEvent(pod *corev1.Pod) error {
+	// the BuildRun name can be empty when the follow was triggered with the buildrun creation
+	if f.buildRunName == "" {
+		f.buildRunName = pod.GetLabels()[buildv1alpha1.LabelBuildRun]
```

Comment:
yeah this setting of buildRunName here will eventually get flagged for a data race either in the unit data race detection or randomly / intermittently with actual usage

OnEvent falls in the same category as Run as I mentioned earlier with respect to multi threadeded 

so change this if block to 

```
brName := f.buildRunName
if len(brName) == 0 {
   brName = pod.GetLabels()[buildv1alpha1.LabelBuildRun]
}
```

And then use `brName` instead of `f.buildRunName` below

Reply:
@gabemontero the file has other places where it accesses `f.buildRunName` but where there is no Pod object at hand to look it up. So, if that approach is not doable, then we need to revisit when the Follower gets instantiated. In `run.go` should we move this to a place after the buildrun was created and its name is known so that it can be assed into `NewFollower` ?